src/container/streamAnalyticsManager.ts

```diff
@@ -74,11 +119,29 @@ export class StreamAnalyticsManager {
                 }
             }
         } catch (error) {
-            error.message = `Cannot parse Stream Analytics publish job information: ${error.message}`;
+            error.message = `Parse Stream Analytics jobs info failed: ${error.message}`;
             throw error;
         }
     }
 
+    private async getJobSubscription(ASAInfo): Promise<AzureSubscription> {
+        await this.azureAccount.waitForFilters();
+        for (const azureSubscription of this.azureAccount.subscriptions) {
+            if (ASAInfo.ASAJobResourceId.indexOf(azureSubscription.subscription.id) >= 0) {
+                return azureSubscription;
+            }
+        }
+
+        const ASAJobName: string = ASAInfo.ASAJobResourceId.split("/").pop();
+        throw new Error(`Cannot find Stream Analytics Job '${ASAJobName}' in your Azure account, please make sure to login to the right acount.`);
+    }
+
+    private async getJobInfoFromDeploymentTemplate(templateFile: string, moduleName) {
+        const templateJson = await fse.readJson(templateFile);
+        const ASAInfo = templateJson.modulesContent[moduleName]["properties.desired"];
```

Comment:
if module is not correctly configured. what will happen? like it missing field of asainfo, or other entries

Reply:
Thanks, I think maybe I should add try catch to handle these errors